pinot-spi/src/main/java/org/apache/pinot/spi/utils/validator/BaseJsonValidator.java

```diff
@@ -0,0 +1,56 @@
+package org.apache.pinot.spi.utils.validator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.fge.jackson.JsonLoader;
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import com.github.fge.jsonschema.core.report.ProcessingMessage;
+import com.github.fge.jsonschema.core.report.ProcessingReport;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.pinot.spi.utils.JsonUtils;
+
+
+abstract public class BaseJsonValidator implements JsonValidator {
+
+  protected JsonSchema _schema;
+  protected ProcessingReport _report;
+
+  protected void loadSchema(String path) {
+    try {
+      JsonNode node = JsonLoader.fromResource(path);
+      JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
+      _schema = factory.getJsonSchema(node);
+    } catch (IOException | ProcessingException e) {
+      e.printStackTrace();
```

Comment:
Avoid using printStackTrace, you can use a logger or simply throw the exception to the caller. Same comment for all such instances